### ISSUE TITLE
Error on call .track() Invalid argument: Instance of 'Object'

### ISSUE THREAD

**Issue by devinyonas** [mixpanel/mixpanel-flutter#153]
Hi, 

When i call:
```dart
await _mixpanel.track(
        name,
        properties: {'relationship': Relationship.mother},
      );
```

I got error: `Invalid argument: Instance of 'Relationship'`

Relationship is an enum, ex:
```dart
enum Relationship {
  mother,
  father;
 }
```

i see that properties accept  `Map<String, dynamic>?` but I'm not sure why its returning `Invalid-argument`

Is `.track()` `properties`  doesn't support custom object?

**Comment by CH1NRU5T** [comment]
### TLDR; Dart is not able to serialize the `Object` of a class.

## Potential fix (1)
You can pass the data like this
```dart
await _mixpanel.track(
        name,
        properties: {'relationship': Relationship.mother.name},
      );
```

## Potential fix (2)
You can define a `toString()` method in your enum and call it while passing the properties
```dart
enum Relationship {
  mother,
  father;

  @override
  String toString() {
    return name;
  }
}
```
```dart
await _mixpanel.track(
        name,
        properties: {'relationship': Relationship.mother.toString()},
      );
```


# Explanation
What I observed by going deep into the package code was that this package uses [Method channels](https://api.flutter.dev/flutter/services/MethodChannel-class.html) to execute code on the platform.

You can see [this](https://github.com/mixpanel/mixpanel-flutter/blob/765acc8d0f0c83ad6cf991441e6e6be54d652da4/lib/mixpanel_flutter.dart#L193) method.

```dart
await _channel.invokeMethod<void>('track',
          <String, dynamic>{'eventName': eventName, 'properties': properties});
```

Now when we call `.invokeMethod` the arguments to this function get serialized by a custom [MessageCodec](https://api.flutter.dev/flutter/services/MessageCodec-class.html)

[Mixpanel's MessageCodec](https://github.com/mixpanel/mixpanel-flutter/blob/main/lib/codec/mixpanel_message_codec.dart) only handles `Uri` & `DateTime` types and uses the parent to handle other types.

The parent [StandardMessageCodec](https://api.flutter.dev/flutter/services/StandardMessageCodec-class.html) only handles well known types like bool, nums, strings etc. 

In short, it does not handle the `Object` of a class or enum.

That is why you get `Invalid argument: Instance of 'Relationship'`

**Comment by devinyonas** [comment]
Hi @CH1NRU5T, thank you for your explanation.
You are right the issue is with serializing in the method channel. 

I have been using `.toString()` like your potential fix (2) since then.
```dart
await _mixpanel.track(
        name,
        properties: {'relationship': Relationship.mother.toString()},
      );
```

will close this issue
